lib/rules/implicit-arrow-linebreak.js

```diff
@@ -109,18 +108,13 @@ module.exports = {
             let followingBody = arrowBody;
             let currentArrow = arrow;
 
-            while (currentArrow) {
+            while (currentArrow && followingBody.value !== "{") {
```

Comment:
I'm confused by this line. According to the JSDoc comment, isn't `arrowBody` an AST node rather than a token? I don't understand how it would have a `value` property.

Reply:
Changed the `arrowBody` param type to be `{ASTNode|Token}`. Thank you for clarifying!

Reply:
Thanks for making that change. I suppose my question is a bit more general -- what are `followingBody` and `arrowBody` actually supposed to be? From the name `arrowBody` I would expect it to be the `body` of the AST node for the arrow function, or perhaps the first token of that AST node. But if I'm understanding correctly, the value might be something before the body, and `followingBody` only reaches the first token of the body (`{`) at the end of the loop. Is `arrowBody` actually just the first token after the arrow token?

Reply:
`followingBody` and `arrowBody` _should_ refer to the body of the arrow function. Prior to the new changes, this wasn't the case -- thank you for asking these questions, as the new changes ensure they are as expected.